homeassistant/components/zha/entity.py

```diff
@@ -3,87 +3,76 @@
 
-from .core.const import (
-    ATTR_MANUFACTURER,
-    ATTR_MODEL,
-    DOMAIN,
-    SIGNAL_GROUP_ENTITY_REMOVED,
-    SIGNAL_GROUP_MEMBERSHIP_CHANGE,
-    SIGNAL_REMOVE,
-)
-from .core.helpers import LogMixin, get_zha_gateway
-
-if TYPE_CHECKING:
-    from .core.cluster_handlers import ClusterHandler
-    from .core.device import ZHADevice
+from .const import DOMAIN
+from .helpers import SIGNAL_REMOVE_ENTITIES, EntityData, convert_zha_error_to_ha_error
 
 _LOGGER = logging.getLogger(__name__)
 
-ENTITY_SUFFIX = "entity_suffix"
-DEFAULT_UPDATE_GROUP_FROM_CHILD_DELAY = 0.5
-
 
-class BaseZhaEntity(LogMixin, entity.Entity):
-    """A base class for ZHA entities."""
-
-    _unique_id_suffix: str | None = None
-    """suffix to add to the unique_id of the entity. Used for multi
-       entities using the same cluster handler/cluster id for the entity."""
+class ZHAEntity(LogMixin, RestoreEntity, Entity):
+    """ZHA eitity."""
 
     _attr_has_entity_name = True
     _attr_should_poll = False
+    remove_future: asyncio.Future[Any]
 
-    def __init__(self, unique_id: str, zha_device: ZHADevice, **kwargs: Any) -> None:
+    def __init__(self, entity_data: EntityData, *args, **kwargs) -> None:
         """Init ZHA entity."""
-        self._unique_id: str = unique_id
-        if self._unique_id_suffix:
-            self._unique_id += f"-{self._unique_id_suffix}"
-        self._state: Any = None
-        self._extra_state_attributes: dict[str, Any] = {}
-        self._zha_device = zha_device
+        super().__init__(*args, **kwargs)
+        self.entity_data: EntityData = entity_data
         self._unsubs: list[Callable[[], None]] = []
 
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return self._unique_id
+        if self.entity_data.entity.icon is not None:
+            # Only custom quirks will realistically set an icon
+            self._attr_icon = self.entity_data.entity.icon
 
-    @property
-    def zha_device(self) -> ZHADevice:
-        """Return the ZHA device this entity is attached to."""
-        return self._zha_device
+        meta = self.entity_data.entity.info_object
+        self._attr_unique_id = meta.unique_id
+
+        if meta.translation_key is not None:
+            self._attr_translation_key = meta.translation_key
+        elif meta.fallback_name is not None:
+            # Only custom quirks will create entities with just a fallback name!
+            #
+            # This is to allow local development and to register niche devices, since
+            # their translation_key will probably never be added to `zha/strings.json`.
+            self._attr_name = meta.fallback_name
+
+        if meta.entity_category == ZHAEntityCategory.CONFIG:
+            self._attr_entity_category = EntityCategory.CONFIG
+        elif meta.entity_category == ZHAEntityCategory.DIAGNOSTIC:
+            self._attr_entity_category = EntityCategory.DIAGNOSTIC
```

Comment:
In this case we probably can just try to initialize an `EntityCategory` directly?

```
self._attr_entity_category = EntityCategory(meta.entity_category)
```

Reply:
I think the intention here was to not mix ZHA and HA entity categories, even though they're "the same".
But yeah, it would work.

Reply:
Yeah, it is the same result, more of an aesthetic nit-pick. Additionally, if there were an unexpected upstream value, it would blow up.

Reply:
done